### PR TITLE
feat: Feat/media pipe learning

### LINKED ISSUE #117: Adapt MediaPipe to project
Understand the code
Run it on your own images and videos
Adapt the code so that the positions of joints get stored in a separate (csv, json) file

**Phase 1: React (Vercel)**

The user's browser does the heavy lifting.

Use the detector.estimatePoses(video) method from MediaPipe in your React code.

Filter the results to only include the keypoints needed for a squat (hips, knees, ankles, heels).

Send a clean JSON array of these coordinates to your Python backend. One array of the 3D and one with the 2D coordinates.

**Phase 2: Python (Render)**

The backend remains "thin" and fast because it doesn't process images, only numbers.

Math/Logic: Calculate the angle of the knee:
$$\theta = \arccos \left( \frac{a^2 + b^2 - c^2}{2ab} \right)$$

(Where a, b, and c are the lengths of the segments between hip, knee, and ankle).

ML Model (PyTorch): Run a very small PyTorch model (<1 MB) on these angles to classify the squat as "Deep," "Shallow," or "Invalid."

This should be tested within the src/ml-research/a8/ directory

### LINKED ISSUE #123: Kinect data normalization
Normalize the Kinect data files in src/data/kinect_* directories to be compatible with the MediaPipe data that will be stored in DB after video, image upload and after live webcam feed

### LINKED ISSUE #130: Connect with DL model

### LINKED ISSUE #132: Update the classification of squats
Update the backend call from the current Invalid, Shallow or Deep classifier to use the one hosted on DagsHub which is Z_MODEL_URI_PROD= and Z_MODEL_URI_DEV= entries in .env.example and make it accesible as a new endpoint as we currently do with Score prediction and weaklink classification but with the z-predictor which is a model that predicts 3D z value based on x and y coords and this is to ligthn the load on the backend and to not require the pytorch/torch package for the backend to avoid bloating it

### PR DESCRIPTION
**Pull request by SamuelFredricBerg**
## Description

Closes #117 
Closes #123 
Closes #130 
Closes #132 
Implements MediaPipe to our webservice as well as the keypoints storage in a DB hosted on SupaBase.

## Type of Change

- [x] New Feature (ML Model/API Endpoint)
- [x] Bug Fix
- [x] Refactoring / Tech Debt
- [x] Documentation Update

## Checklist

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] (ML) I have verified model performance metrics.
- [x] (Backend) I have updated the API documentation (OpenAPI).
- [x] Tests pass locally, and new tests have been added where applicable.
- [x] I have added necessary documentation (if appropriate).

### CODE REVIEW

File: src/backend/app/services/model_service.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def _init_mlflow() -> str:
     return uri
 
 
-# * Current "entry point"
+# TODO: Unused???
 def _model_name_for_variant(variant: str) -> str:
```
[the remark this answers is not in the document]
Comment (SamuelFredricBerg) [reply src/backend/app/services/model_service.py:127]: Legacy code

**Comment by SamuelFredricBerg** [comment]
@copilot exclude the 2d data from storing and saving to csv in the frontend SquatAnalyzer file foremost

**Comment by SamuelFredricBerg** [comment]
@copilot Are the normalized kinect data accurate to be used to train DL model with 3d data gathered from MediaPipe?

**Comment by SamuelFredricBerg** [comment]
@copilot For training a DL model to estimate z axis value from x and y is the Kinect data normalized in a way to be compatible with the 3d data from MediaPipe

**Comment by SamuelFredricBerg** [comment]
@copilot So for this example MediaPipe data would the z axis value need to be flipped in the normalization?
if yes implement it.
```
left_shoulder_3d_x,left_shoulder_3d_y,left_shoulder_3d_z,right_shoulder_3d_x,right_shoulder_3d_y,right_shoulder_3d_z,left_elbow_3d_x,left_elbow_3d_y,left_elbow_3d_z,right_elbow_3d_x,right_elbow_3d_y,right_elbow_3d_z,left_hip_3d_x,left_hip_3d_y,left_hip_3d_z,right_hip_3d_x,right_hip_3d_y,right_hip_3d_z,left_knee_3d_x,left_knee_3d_y,left_knee_3d_z,right_knee_3d_x,right_knee_3d_y,right_knee_3d_z,left_ankle_3d_x,left_ankle_3d_y,left_ankle_3d_z,right_ankle_3d_x,right_ankle_3d_y,right_ankle_3d_z,left_heel_3d_x,left_heel_3d_y,left_heel_3d_z,right_heel_3d_x,right_heel_3d_y,right_heel_3d_z,left_foot_index_3d_x,left_foot_index_3d_y,left_foot_index_3d_z,right_foot_index_3d_x,right_foot_index_3d_y,right_foot_index_3d_z
0.1564,-0.0759,-0.3884,-0.1305,-0.1673,-0.3627,0.1088,0.1482,-0.2880,-0.2170,-0.0674,-0.1829,0.1042,0.0110,-0.0103,-0.1050,-0.0118,0.0110,0.1798,0.3697,-0.0278,-0.2091,0.3664,-0.0192,0.1535,0.6979,0.1189,-0.1553,0.6853,0.1569,0.1521,0.7425,0.1247,-0.1490,0.7232,0.1702,0.1826,0.8183,0.0105,-0.1909,0.8114,0.0681
```

**Comment by SamuelFredricBerg** [comment]
@copilot fix the errors in SquatAnalyzer.jsx according to previous request to remove 2d related aspects

**Comment by SamuelFredricBerg** [comment]
@copilot Update documentastion files to represent the changes made with this PR

**Comment by SamuelFredricBerg** [comment]
@copilot Update docs to include latest updates, CHANGELOG, DEPLOYMENT, READMEs

**Review by LNU4 — APPROVED** [review]
LGTM, skipped the datasets
